File: 0238-product-of-array-except-self/0238-product-of-array-except-self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        ans=[]
        x=1
        for i in range(len(nums)):
            if i==0:
                ans.append(x)
            else:
                x=x*nums[i-1]
                ans.append(x)
        x=1
        for i in range(len(nums)-1,-1,-1):
            if i==len(nums)-1:
                continue
            else:
                x=x*nums[i+1]
                ans[i]=ans[i]*x
        return ans

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int> ans;
        int x=1;
        for(int i=0;i<nums.size();i++){
            if(i==0){
                ans.push_back(x);
            }
            else{
                x=x*nums[i-1];
                ans.push_back(x);
            }
        }
        x=1;
        for(int j=nums.size()-1;j>=0;j--){
            if(j==nums.size()-1){
                continue;
            }
            else{
                x=x*nums[j+1];
                ans[j]=ans[j]*x;
            }
        }
        return ans;
    }
};